tools/scripts/cn_metrics.py

```diff
@@ -69,6 +74,14 @@ def full_tree(ybuf: Optional[OrderedDict[str, Any]], opt):
     if ybuf["info"]["open"] == False:
         return
 
+    # TODO: How to test if ybuf[] contain at least one "nodes" stanza?
+    #
+    try:
+        for node in ybuf["nodes"]:
```

Comment:
`if "nodes" in ybuf`?

Reply:
ybuf is some kind of monstrosity i don't grok, it returns NoneType if "nodes" is not in ybuf, but you shouldn't test explicitly for None/NoneType, not to mention the none stuff seems to vary depending on Python version, uhg...

Reply:
"nodes" in ybuf

tells you that there is a key called "nodes". If you are looking to make sure there is a node in the array, then do this:

`if ybuf.get("nodes", []): return`